torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8479,29 +8457,143 @@ def wrapper(t: torch.Tensor, *args, dim=None, keepdim=False, **kwargs):
 
 # Reduction operator database
 reduction_op_db: List[ReductionOpInfo] = [
+    ReductionOpInfo(
+        'amax',
+        dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+        sample_inputs_func=sample_inputs_amax_amin,
+        reference=reference_reduction_numpy(np.amax),
+        skips=(
+            # FIXME: dim=[] is incorrectly reducing all dimensions
+            SkipInfo('TestReductions', 'test_dim_empty'),
+            SkipInfo('TestReductions', 'test_dim_empty_keepdim'),
+        ),
+    ),
+    ReductionOpInfo(
+        'amin',
+        dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+        sample_inputs_func=sample_inputs_amax_amin,
+        reference=reference_reduction_numpy(np.amin),
+        skips=(
+            # FIXME: dim=[] is incorrectly reducing all dimensions
+            SkipInfo('TestReductions', 'test_dim_empty'),
+            SkipInfo('TestReductions', 'test_dim_empty_keepdim'),
+        ),
+    ),
+    ReductionOpInfo(
+        'all',
+        identity=True,
+        nan_policy='omit',
+        supports_multiple_dims=False,
+        supports_autograd=False,
+        result_dtype=torch.bool,
+        dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+        reference=reference_reduction_numpy(np.all),
+        skips=(
+            # FIXME: does not support passing keepdim without dim
+            SkipInfo('TestReductions', 'test_dim_default_keepdim'),
+            # FIXME: does not support dim=None
+            SkipInfo('TestReductions', 'test_dim_none'),
+            SkipInfo('TestReductions', 'test_dim_none_keepdim'),
+            # FIXME: uint8 input returns uint8 instead of bool
+            SkipInfo('TestReductions', 'test_result_dtype'),
+            SkipInfo('TestReductions', 'test_ref_random_input_small', dtypes=[torch.uint8]),
+            # FIXME: out is not support when calling without dim and keepdim
+            SkipInfo('TestCommon', 'test_out'),
+        ),
+    ),
+    ReductionOpInfo(
+        'any',
+        identity=False,
+        nan_policy='omit',
```

Comment:
This is incorrect! all/any treat np.nan as True. Maybe add a policy 'ignore' that does not do anything special with np.nan and just passes it through to the operation?

Reply:
Or a policy 'largest' that treats NaN as the largest value, which also means non-zero and thus True? This would be inline with our sorting routine which sorts NaN as the largest value.